editor/code_editor.h

```diff
@@ -30,19 +30,80 @@
 
 #pragma once
 
+#include "core/math/vector2i.h"
+#include "core/object/script_language.h"
 #include "scene/gui/box_container.h"
 #include "scene/gui/button.h"
 #include "scene/gui/check_box.h"
 #include "scene/gui/code_edit.h"
 #include "scene/gui/dialogs.h"
 #include "scene/gui/label.h"
+#include "scene/gui/panel_container.h"
 #include "scene/gui/popup.h"
+#include "scene/gui/rich_text_label.h"
 #include "scene/main/timer.h"
 
 class MenuButton;
 class CodeTextEditor;
 class LineEdit;
 
+class RefactorRenamePopup : public PanelContainer {
```

Comment:
Is there a reason that this isn't a Popup / PopupPanel?
The functionality is similar and it should be on top, so it doesn't get cut off:

![image](https://github.com/user-attachments/assets/8f25d84f-162e-4d57-ac81-a4db7e4bcdde)

Reply:
Ah, I remember. If it's a popup, it doesn't follow the code scrolling. I think the box following its source is really important.

I'll try though to calculate the position of the fake popup better.